mycore-base/src/main/java/org/mycore/common/xml/MCRXMLFunctions.java

```diff
@@ -860,6 +861,44 @@ public static String stripHtml(final String s) {
         return StringEscapeUtils.unescapeHtml4(res.toString()).replaceAll(TAG_SELF_CLOSING, "");
     }
 
+    /**
+     * Returns the x-mapping for the given mapping prefix.
+     * <br/>
+     * Example: <code>getXMapping("mir_genres", "article", "schemaOrg")</code> will return <code>ScholarlyArticle</code>
+     *
+     * @param classificationId
+     * @param categoryId
+     * @param mappingPrefix
+     *
+     * @return the mapping or an empty string when there is no such mapping for the given parameters
+     * */
+    public static String getXMapping(String classificationId, String categoryId, String mappingPrefix) {
+        MCRCategory category = null;
+        try {
+            MCRCategoryID categID = MCRCategoryID.fromString(classificationId + ":" + categoryId);
+            MCRCategoryDAO dao = MCRCategoryDAOFactory.getInstance();
+            category = dao.getCategory(categID, 0);
+            Optional<MCRLabel> label = category.getLabel("x-mapping");
+            if (label.isEmpty()) {
+                return "";
+            }
+            MCRLabel xmapping = label.get();
+            if (!xmapping.getText().contains(mappingPrefix)) {
+                return "";
+            }
+
+            Optional<String> result = Arrays.stream(xmapping.getText().split(" "))
+                .filter(mapping -> mapping.startsWith(mappingPrefix))
+                .findFirst();
+
+            return result.isPresent() ? result.get().split(":")[1] : null;
```

Comment:
Null as return does not match the Javadoc or is inconsistent compared to the empty string as return.